Switching to a similar workflow as stm32-rs, expanded the readme

allexoll opened the pull request:
Hey everyone, after a while working on this, i think it would make more sense to use a similar workflow as they do over at @stm32-rs.  I've switched lpc546xx-pac to it (with a few tweaks to stick to the lpcxxxxx-pac naming. 

I've expanded the readme as well, and i hope that some day we can switch the other pacs this way too. It makes it easier to track svd patches (the UART permission fix is present in lpx845 too for example) and share them across different families. 

There's still the discussion about versions to have, but this PR does not impact it (as the resulting lpc546xx-pac is identical to the one existing before)

If someone could review this, i can then merge it.

Remarks welcome obviously

alexismarquet committed:
Switching to a similar workflow as stm32-rs, expanded the readme

hannobraun reviewed (commented):
I've taken a quick look, which is all I can manage right now. Reviewing the whole diff is pretty much impossible anyway. I think it would be advisable to separate manual changes and automated changes to generated code into separate commits, in the the future.

I'm all for reusing the stm32-rs infrastructure. No objections here.

allexoll commented:
Thanks

Then a possible option would be moving to a `lpc-pac-nightlies` repository as well. I'll look into how it's done over there to see what that would entail.

alexismarquet committed:
switching to github actions since travis is not available anymore

alexismarquet committed:
fix type in cmsis

alexismarquet committed:
updated the makefile to not need a rename afterwards. fixed typos in doc

alexismarquet committed:
added lpc81x

alexismarquet committed:
fixed selection of family when running script

alexismarquet committed:
added jobs

allexoll commented:
All right, so i had another look at this, and i do think moving to a `nightlies` repository as well would be best. This PR should bring it in that direction:

* Switched to Github Actions since travis is not supporting OS projects
* GA will build/check pacs on push/PR
* GA will push built crates onto a future `lpc-pac-nightlies`
* GA will push memory maps onto a future `lpc-pac-mmaps` and comment the comparison on PRs
* I edited the makefile for the `x*-pac` format we are using here for LPC chips
* I added the LPC81x family to show that it also works with more than one family but i don't really intend to publish it to crates.io since I'm not familiar with the device


Once this PR is merged, all that should be needed is to add the secrets needed for all the GA to this repository, create `lpc-pac-nightlies` and `lpc-pac-mmap` and we should be good to go. the 5000+ files changes is the deletion of all the rs files in lpc546xx-pac since it will move to `lpc-pac-nightlies`.

alexismarquet committed:
fixed gitignore and expanded README
